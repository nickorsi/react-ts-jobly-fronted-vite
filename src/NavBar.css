.navbar {
  background-image: linear-gradient(hsl(0, 0%, 100%), hsl(0, 0%, 90%));
  position: fixed;
  width: 100%;
  z-index: 1000;
}

.navbar div {
  display: flex;
  justify-content: space-between;
}

.navbar a {
  color: black;
  text-decoration: none;
}

.navbar a:hover {
  text-decoration: underline;
}

.navbar a:active {
  text-decoration: none;
}

.navbar a.active {
  font-weight: bold;
}

.nav-item {
  font-size: 1.1rem;
}