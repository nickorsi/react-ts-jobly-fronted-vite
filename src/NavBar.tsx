import React from "react";
import { NavLink } from "react-router-dom";

function NavBar() {
    return (
        <nav>
            <NavLink
                to='/'
                className={({ isActive, isPending }) =>
                    isPending ? "pending" : isActive ? "active" : ""
                }
            >
                Jobly
            </NavLink>
            <NavLink
                to='/companies'
                className={({ isActive, isPending }) =>
                    isPending ? "pending" : isActive ? "active" : ""
                }
            >
                Companies
            </NavLink>
            <NavLink
                to='/jobs'
                className={({ isActive, isPending }) =>
                    isPending ? "pending" : isActive ? "active" : ""
                }
            >
                Jobs
            </NavLink>
        </nav>
    )
}

export default NavBar;