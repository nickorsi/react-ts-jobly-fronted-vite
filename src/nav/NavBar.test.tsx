import {describe, expect, it} from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import NavBar from "./NavBar.tsx";

describe("Simple Render Tests", () => {
    it("renders without crashing", function () {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );
    });
    it("matches the snapshot", function () {
        const { container } = render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
    it("renders with the correct content", function () {
        const { container } = render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        );

        const links = container.querySelectorAll("a");

        expect(links.length).toEqual(3);
        expect(screen.getByText("Jobly ðŸ’¼").getAttribute('href')).toEqual('/');
        expect(
            screen.getByText("Companies").getAttribute('href')
        ).toEqual('/companies');
        expect(screen.getByText("Jobs").getAttribute('href')).toEqual('/jobs');
    });
});
