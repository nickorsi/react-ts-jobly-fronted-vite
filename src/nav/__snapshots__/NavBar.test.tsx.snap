// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Render Tests > matches the snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg bg-body-tertiary"
  >
    <div
      class="container-fluid"
    >
      <a
        aria-current="page"
        class="active navbar-brand ms-3"
        data-testid="home-nav-button"
        href="/"
      >
        Jobly ðŸ’¼
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          class="navbar-nav ms-auto mb-2 mb-lg-0"
        >
          <li
            class="nav-item mx-5"
          >
            <a
              class=""
              data-testid="companies-nav-button"
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item mx-5"
          >
            <a
              class=""
              data-testid="jobs-nav-button"
              href="/jobs"
            >
              Jobs
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
